bool obstacle
float32[] index
